### PR TITLE
Discover iaas resource vm as Configured system.

### PR DESCRIPTION
**Pull request by kenneth88888**
Discover Managed Services iaas virtual_machine resources as Configured Systems in ManageIQ.

### CODE REVIEW

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb [hunk 1]
```diff
@@ -14,17 +14,37 @@ def configuration_profiles
...
+    collector.virtual_machines.each do |virtual_machine|
       persister.configured_systems.build(
-        :manager_ref           => stack["id"].to_s,
-        :hostname              => stack["name"],
-        :last_checkin          => stack["created_at"],
-        :build_state           => stack["build"],
-        :mac_address           => stack["status"],
-        :configuration_profile => persister.configuration_profiles.lazy_find(template_ref)
+        :manager_ref          => virtual_machine["id"].to_s,
+        :name                 => virtual_machine["name"] + ' - ' + virtual_machine["ipaddresses"].to_s,
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:45]: Why tack on the IP addresses to the name?

Comment (kenneth88888) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:45]: Name we have today is not unique to identify the vm(something like - aws_instance.web-server). We can look into returning hostname or some other more unique name from CAM side.

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb [hunk 2]
```diff
@@ -14,17 +14,37 @@ def configuration_profiles
...
+        :ipaddress            => virtual_machine["ipaddresses"].to_s,
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:46]: So this is expecting a single ip address, probably should be `virtual_machine["ipaddresses"]&.first`

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb (hunk 1 above)
Comment (agrare) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:45]: Okay, name doesn't have to be unique so unless it is common to have lots of IaaS resources with the same name we don't need to try to make each name unique here.  Ideally we should show the name which they'd see in CAM so there is consistency.

**Comment by agrare** [comment]
@kenneth88888 I see that you switched from collecting stacks to collecting IaaS resources, what is the rationale for the switch?  No issue just curious

**Comment by kenneth88888** [comment]
> @kenneth88888 I see that you switched from collecting stacks to collecting IaaS resources, what is the rationale for the switch? No issue just curious

@agrare we wanted to move the Stack to orchestrator stack at some point and only show vm's as configured systems.

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb (hunk 2 above)
Comment (kenneth88888) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb]: Addressed this in latest PR

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb (hunk 1 above)
Comment (kenneth88888) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb]: Addressed this in latest PR

Comment (agrare) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb]: :+1:

**Pull request merged**